File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/ClearUserSessionBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.serialize

internal class ClearUserSessionBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    companion object {
        private const val clearUserSessionErrorCode =
            "usercentrics_flutter_clearUserSession_error"
    }

    override val name: String
        get() = "clearUserSession"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        usercentrics.instance.clearUserSession(
            onSuccess = {
                result.success(it.serialize())
            },
            onError = {
                result.error(
                    clearUserSessionErrorCode,
                    it.message,
                    it
                )
            },
        )
    }
}

File: ios/Classes/Bridge/ClearUserSessionBridge.swift
import Foundation

struct ClearUserSessionBridge : MethodBridge {
    
    let name: String = "clearUserSession"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        
        usercentrics.shared.clearUserSession { status in
            result(status.serialize())
        } onError: { error in
            result(FlutterError(code: "usercentrics_flutter_clearUserSession_error",
                                message: error.localizedDescription,
                                details: nil ))
        }
    }
}
